### DISCUSSION TITLE
Symfony Deprecations - 2 only

### DISCUSSION THREAD

**Discussion by Allislove** [community/community#21446] [Other Feature Feedback, Questions, & Ideas]
<ol>
<li>User Deprecated: Using Groups is deprecated since version 2.2 and will be removed in 3.0.</li>
<li>The body_listener config has been enabled by default and will be disabled by default in FOSRestBundle 3.0. Please enable or disable it explicitly.</li>
</ol>
<p><strong>Im upgrading Symfony 3.0 to 3.4</strong></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/e/0e5f8480a1aefa0e2664f5c4ff1aa638b6032548.png" data-download-href="/uploads/short-url/239ksNW3opvSKc8d9Fh2GUpwosg.png?dl=1" title="Captura de pantalla de 2021-07-26 17-50-15" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/54194376/181064589-caa1cc74-be83-4623-925b-107f9823ea68.png" alt="Captura de pantalla de 2021-07-26 17-50-15" data-base62-sha1="239ksNW3opvSKc8d9Fh2GUpwosg" width="690" height="389" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/e/0e5f8480a1aefa0e2664f5c4ff1aa638b6032548_2_690x389.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/e/0e5f8480a1aefa0e2664f5c4ff1aa638b6032548_2_1035x583.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/0/e/0e5f8480a1aefa0e2664f5c4ff1aa638b6032548.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/e/0e5f8480a1aefa0e2664f5c4ff1aa638b6032548_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Captura de pantalla de 2021-07-26 17-50-15</span><span class="informations">1343×758 65.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I have resolved all the previous warnings, and they were many! But I have not been able to give a solution to these two! In other words, it is too difficult for me to find out where the fault is, to find a solution.</p>

**Comment by Zerotask** [comment]
<p>You should at least provide the source code, otherwise we can’t help you.</p>

**Comment by Allislove** [comment, marked as answer]
<p>This is my composer.json</p>
<pre><code class="lang-auto">{
    "name": "project-namehere",
    "license": "proprietary",
    "type": "project",
    "autoload": {
        "psr-4": {
            "": "src/"
        },
        "classmap": [
            "app/AppKernel.php",
            "app/AppCache.php"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "require": {
        "php": "&gt;=5.5.9",
        "symfony/symfony": "3.4.*",
        "doctrine/orm": "^2.5",
        "doctrine/doctrine-bundle": "^1.6",
        "doctrine/doctrine-cache-bundle": "^1.2",
        "symfony/swiftmailer-bundle": "^2.3",
        "symfony/monolog-bundle": "^3.1",
        "symfony/polyfill-apcu": "^1.0",
        "sensio/distribution-bundle": "^5.0",
        "sensio/framework-extra-bundle": "^3.0.2",
        "incenteev/composer-parameter-handler": "^2.0",
        "friendsofsymfony/user-bundle": "~2.0@dev",
        "tetranz/select2entity-bundle": "2.*",
        "friendsofsymfony/rest-bundle": "^2.1",
        "nelmio/api-doc-bundle": "^2.13",
        "pagomio/pagomio-sdk-php": "^1.0",
        "nelmio/cors-bundle": "^1.5",
        "symfony/apache-pack": "^1.0"
    },
    "require-dev": {
        "sensio/generator-bundle": "^3.0",
        "symfony/phpunit-bridge": "^3.0"
    },
    "scripts": {
        "symfony-scripts": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ],
        "post-install-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ],
        "post-update-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ]
    },
    "extra": {
        "symfony-app-dir": "app",
        "symfony-bin-dir": "bin",
        "symfony-var-dir": "var",
        "symfony-web-dir": "web",
        "symfony-tests-dir": "tests",
        "symfony-assets-install": "relative",
        "incenteev-parameters": {
            "file": "app/config/parameters.yml"
        },
        "require": "3.4.*"
    }
}

</code></pre>
<p><strong>And this is a log from the (1) first error!</strong></p>
<pre><code class="lang-auto">User Deprecated: Using Groups is deprecated since version 2.2 and will be removed in 3.0.
Hide context    Hide trace
[▼
  "exception" =&gt; ErrorException {#392 ▶}
]
{▼
  /home/iamother/Php/tcl-servicios/vendor/friendsofsymfony/user-bundle/Model/GroupableInterface.php:14 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/Debug/DebugClassLoader.php:152 {▶}
  Symfony\Component\Debug\DebugClassLoader-&gt;loadClass() {}
  /home/iamother/Php/tcl-servicios/vendor/friendsofsymfony/user-bundle/Model/User.php:24 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/Debug/DebugClassLoader.php:152 {▶}
  Symfony\Component\Debug\DebugClassLoader-&gt;loadClass() {}
  /home/iamother/Php/tcl-servicios/src/AppBundle/Entity/User.php:15 {▼
    ›  */
    › class User extends BaseUser
    › {
  }
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/Debug/DebugClassLoader.php:152 {▶}
  Symfony\Component\Debug\DebugClassLoader-&gt;loadClass() {}
  spl_autoload_call() {}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/Security/Core/Authentication/Token/UsernamePasswordToken.php:105 {▶}
  Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken-&gt;unserialize() {}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/Security/Http/Firewall/ContextListener.php:257 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/Security/Http/Firewall/ContextListener.php:97 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Bundle/SecurityBundle/Debug/WrappedListener.php:46 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Bundle/SecurityBundle/Debug/TraceableFirewallListener.php:35 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/Security/Http/Firewall.php:84 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Bundle/SecurityBundle/EventListener/FirewallListener.php:48 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/Debug/WrappedListener.php:115 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php:214 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php:44 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/Debug/TraceableEventDispatcher.php:143 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:127 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68 {▶}
  /home/iamother/Php/tcl-servicios/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:200 {▶}
  /home/iamother/Php/tcl-servicios/web/app_dev.php:28 {▶}
}
</code></pre>
<p><strong>Log for the 2 Error</strong></p>
<pre><code class="lang-auto">The body_listener config has been enabled by default and will be disabled by default in FOSRestBundle 3.0. Please enable or disable it explicitly.
Hide context    Show trace
[▼
  "exception" =&gt; SilencedErrorContext {#1368 ▼
    +count: 1
    -severity: E_USER_DEPRECATED
    trace: {▶}
  }
]
</code></pre>
